tests/validator_test.go

```diff
@@ -895,6 +908,22 @@ func eventuallyFailToCreateVm(vm *kubevirtv1.VirtualMachine) bool {
 	}, shortTimeout).Should(Equal(metav1.StatusReasonInvalid), "Should have given the invalid error")
 }
 
+func failVmCreationToIncreaseRejectedVmsMetrics() {
+	rejectedCountBefore := totalRejectedVmsMetricsValue()
+	template := TemplateWithRules()
```

Comment:
In the current code the template would not be deleted, if one of the `Expect()` calls panics.

Instead, can you create the template before this function is called, and pass it as a parameter? Then it can be deleted in `AfterEach()`.